src/main/kotlin/org/jitsi/nlj/stats/PacketIOActivity.kt

```diff
@@ -16,14 +16,38 @@
 
 package org.jitsi.nlj.stats
 
-import kotlin.math.max
+import org.jitsi.nlj.util.NEVER
+import org.jitsi.nlj.util.latest
+import java.time.Instant
 
 class PacketIOActivity {
-    var lastPacketReceivedTimestampMs: Long = 0
-    var lastPacketSentTimestampMs: Long = 0
+    var lastRtpPacketReceivedTimestamp: Instant = NEVER
```

Comment:
Do you want to have setters (for this and the other two fields) that don't allow time to go backwards (i.e. that check whether a new value is greater than the old value)?

This could happen, e.g., with oddities of thread scheduling effects.

Reply:
Preventing from going backwards is probably a good idea; I wouldn't expect we'd have issues with going back for a large-enough amount of time that we'd really care, though (and I don't think anything is 'holding on' to a time and then updating this, but could be a nice protection against that).